src/settings/imapimport/AddImapImportWizard.ts

```diff
@@ -0,0 +1,163 @@
+import stream from "mithril/stream"
+import Stream from "mithril/stream"
+import { createWizardDialog, wizardPageWrapper } from "../../gui/base/WizardDialog.js"
+import { assertMainOrNode } from "../../api/common/Env"
+import { EnterImapCredentialsPage, EnterImapCredentialsPageAttrs } from "./EnterImapCredentialsPage.js"
+import { ConfigureImapImportPage, ConfigureImapImportPageAttrs } from "./ConfigureImapImportPage.js"
+import { ImapImportState, ImportState } from "../../api/worker/imapimport/ImapImportState.js"
+import { ImapImportStartedPage, ImapImportStartedPageAttrs } from "./ImapImportStartedPage.js"
+import { TranslationKey } from "../../misc/LanguageViewModel.js"
+import { isDomainName } from "../../misc/FormatValidator.js"
+
+assertMainOrNode()
+
+export type AddImapImportData = {
+	model: ImapImportModel
+}
+
+export interface ImapImportModelConfig {
+	readonly imapAccountHost: string
+	readonly imapAccountPort: string
+	readonly imapAccountUsername: string
+	readonly imapAccountPassword: string
+	readonly rootImportMailFolderName: string
+}
+
+export class ImapImportModel {
+	private _imapAccountHost: stream<string>
+	private _imapAccountPort: stream<string>
+	private _imapAccountUsername: stream<string>
+	private _imapAccountPassword: stream<string>
+	private revealImapAccountPassword: boolean = false
+	private _rootImportMailFolderName: stream<string>
+	private _imapImportState: ImapImportState
+
+	constructor(readonly config: ImapImportModelConfig) {
+		this._imapAccountHost = stream(config.imapAccountHost)
+		this._imapAccountPort = stream(config.imapAccountPort)
+		this._imapAccountUsername = stream(config.imapAccountUsername)
+		this._imapAccountPassword = stream(config.imapAccountPassword)
+		this._rootImportMailFolderName = stream(config.rootImportMailFolderName)
+		this._imapImportState = new ImapImportState(ImportState.NOT_INITIALIZED)
+	}
+
+	get imapAccountHost(): Stream<string> {
```

Comment:
I don't think it makes sense to re-wrap all of them in getters/setters, also I would prefer to avoid streams unless you have a strong reason to use them